### PR TITLE
Optimizer initialisation in Trainer

### LINKED ISSUE #237: change optimizer setting

### PR DESCRIPTION
**Pull request by agsedova**
close #237

### CODE REVIEW

**Comment by agsedova** [comment]
add try-except in initialize_optimizer

**Comment by agsedova** [comment]
Guys, sorry for a misunderstanding today: the idea was that the `initialise_optimizer`function is called in each trainer individually (and not in main Trainer constructor). It is crucial for wrapping the torch model to sklearn model in Cleanlab: the optimizer should be given non-initialized there (and the `initialise_optimizer`function woun't be called there). Pls, check the changes once again.

File: knodle/trainer/config.py [hunk 1]
```diff
@@ -30,6 +31,7 @@ def __init__(
             caching_suffix: str = "",
             saved_models_dir: str = None
     ):
+        self.seed = seed
```
Comment (marina-sp) [inline knodle/trainer/config.py:34]: Do you need this to seed CleanLab? I also wanted to add this line for Snorkel, so nothing against it, just curious

**Review by marina-sp — APPROVED** [review]
As already said, the changes look fine to me.

File: knodle/trainer/config.py (hunk 1 above)
Comment (agsedova) [reply knodle/trainer/config.py:34]: yes! Cleanlab takes seed as an input parameter

**Pull request merged**